compiler/x/codegen/BinaryEvaluator.cpp

```diff
@@ -1937,13 +1937,59 @@ TR::Register *OMR::X86::TreeEvaluator::integerMulEvaluator(TR::Node *node, TR::C
    TR::Node     *firstChild     = node->getFirstChild();
    TR::Node     *secondChild    = node->getSecondChild();
    bool          nodeIs64Bit    = TR::TreeEvaluator::getNodeIs64Bit(node, cg);
+   bool isMemRef                = firstChild->getOpCode().isMemoryReference();
+   TR_X86OpCodes opCode, regOpCode, memOpCode, movOpCode;
    intptrj_t    value;
 
    if (node->isDualCyclic())
       {
       return TR::TreeEvaluator::integerDualMulEvaluator(node, cg);
       }
 
+   // Opcode evaluation for bmul, smul imul and lmul
+   switch (secondChild->getDataType())
+      {
+      case TR::Int8:
+         value = secondChild->getByte();
+         opCode = isMemRef ? IMUL1AccMem : IMUL1AccReg;
+         regOpCode = IMUL1AccReg;
+         memOpCode = IMUL1AccMem;
```

Comment:
I don't see where the register dependencies are setup for these instructions - you need to tell the code generator which value need to be in `ax` for these instructions to work ... I don't see any regdeps being added to specially handle this... Is it done in the generic analyzer? I don't think it would be so please make clear where the dependency is being handled.